com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -268,15 +268,16 @@ internal void ChangeOwnership(NetworkObject networkObject, ulong clientId)
                 throw new SpawnStateException("Object is not spawned");
             }
 
-            networkObject.OwnerClientId = clientId;
+            networkObject.NextOwner = clientId;
+            networkObject.HasNextOwner = true;
 
-            // Server adds entries for all client ownership
-            UpdateOwnershipTable(networkObject, networkObject.OwnerClientId);
+            networkObject.MarkVariablesDirty(true);
```

Comment:
These two are the heavy-lifting part. At next update, the variable will be sent to new owner